modules/dnn/test/test_layers.cpp

```diff
@@ -1764,6 +1764,50 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_ShuffleChannel, Combine(
 /*group*/        Values(1, 2, 3, 6), dnnBackendsAndTargets(/*with IE*/ false)
 ));
 
+TEST(Layer_Test_ReduceMean, accuracy_input_0)
+{
+    vector<int> szData = { 2, 1, 2, 1 ,2 };
+    std::vector<float> initData = { 0, 1, 2, 3, 4, 5, 6, 7 };
+    Mat inpInitA(szData, CV_32FC1, Mat(initData).data);
+    std::vector<float> resAxes0 = { 2, 3, 4, 5 };
+    std::vector<float> resAxes1 = { 0, 1, 2, 3, 4, 5, 6, 7 };
+    std::vector<float> resAxes2 = { 1, 2, 5, 6 };
+    std::vector<float> resAxes3 = { 0, 1, 2, 3, 4, 5, 6, 7 };
+    std::vector<float> resAxes4 = { 0.5, 2.5, 4.5, 6.5 };
+    std::vector < vector<float>> resReduceMean = { resAxes0, resAxes1, resAxes2, resAxes3, resAxes4 };
+
+
+    for (int i = 0; i < resReduceMean.size(); i++)
+    {
+        Net net;
+        LayerParams lp;
+        lp.set("keepdims", 0);
+        lp.type = "Reduce";
+        lp.set("reduce", "MEAN");
+        lp.name = "testReduceMean";
+        lp.set("axes", i);
+        lp.blobs.push_back(inpInitA);
+
+        net.addLayerToPrev(lp.name, lp.type, lp);
+        net.setInput(inpInitA);
+        net.setPreferableBackend(DNN_BACKEND_OPENCV);
+
+        Mat output = net.forward();
+        MatShape gt_shape;
+        for (int j = 0; j < szData.size(); j++)
+        {
+            if (i == j) continue;
+            gt_shape.push_back(szData[j]);
+        }
+
+        EXPECT_EQ(gt_shape, shape(output));
+
+        Mat a = output.reshape(1, output.total());
+        normAssert(a, Mat(resReduceMean[i]).t());
```

Comment:
@fengyuentau @vpisarev @dkurt I have to add `.t()` to have the same shapes. Is it right solution?

Reply:
I guess it relates to 1d mat changes. `resAxesn` is 1d tensor with shape [n], when it is loaded into a Mat, if I remember correctly, it was of shape [n, 1] before and now [1, n] after 1d mat is supported on 5.x.

Reply:
Reshape does not support 1d for now. I filed a bug about it. Also I re-implemented check with 2d case without `.t()` call.

Reply:
`Mat a = output.reshape(output.total(), 1);` also a valid solution

Reply:
@fengyuentau @dkurt I reworked it a bit. Could you look again?

Reply:
Forget my previous comment since it is kind of misleading.

1. Shapes of result and reference are checked in 1803.
2. So lateral is checking values. `Mat a` is still 2d, e.g. [n, 1]. However, loading a vector into Mat using `Mat(resReduceMean[i])`, we now get a Mat of shape [n]. I guess `.t()` creates the extra appended dimension.

Sugguestions to change:

1. Use other asserting function to check value which does not check shape. Or,
2. Drop shape checking at line 1803 and reshaping at line 1805. Initialize the reference Mat with correct shape and compare with output directly:

```.diff
diff --git a/modules/dnn/test/test_layers.cpp b/modules/dnn/test/test_layers.cpp
index e9e93a7853..587c2275a6 100644
--- a/modules/dnn/test/test_layers.cpp
+++ b/modules/dnn/test/test_layers.cpp
@@ -1802,10 +1802,7 @@ TEST(Layer_Test_ReduceMean, accuracy_input_0)
 
         EXPECT_EQ(gt_shape, shape(output));
 
-        // reshape does not support 1d for now: https://github.com/opencv/opencv/issues/25174
-        Mat a = output.reshape(1, output.total());
-        Mat gt = Mat(resReduceMean[i].size(), 1, CV_32FC1, &resReduceMean[i][0]);
-        normAssert(gt, a);
+        normAssert(output, Mat(gt_shape, CV_32F, resReduceMean[i].data()));
     }
 }
```

Reply:
Done!